utils/lib/InstallPlan.ts

```diff
@@ -0,0 +1,173 @@
+import * as fs from 'fs';
+import * as yaml from 'js-yaml';
+
+import Component from './Component';
+import { removeBasePath } from './classHelpers';
+import { fetchNRGraphqlResults } from './nr-graphql-helpers';
+import { INSTALL_PLAN_MUTATION } from '../constants';
+
+import type {
+  InstallPlanConfig,
+  InstallPlanConfigTargetOS,
+  InstallPlanInstall,
+} from '../types/InstallPlanConfig';
+import type {
+  InstallPlanDirectiveInput,
+  InstallPlanMutationVariable,
+  InstallPlanTargetInput,
+} from '../types/InstallPlanMutationVariables';
+
+interface InstallPlanMutationResponse {
+  installPlanStep: {
+    id: string;
+  };
+}
+
+class InstallPlan extends Component<InstallPlanConfig, string> {
+  /**
+   * @returns Filepath for the configuration file (from top-level directory).
+   */
+  getConfigFilePath() {
+    // Lines next few lines are a hack to allow us to pass in the install plan id
+    // as the localPath, this code then parses all install plans and sets
+    // the correct path to the requested install plan
+    const id = this.localPath;
+    const allInstallPlans = getAllInstallPlanFiles(this.basePath).map((p) => ({
+      filePath: p,
+      content: yaml.load(
+        fs.readFileSync(p).toString('utf-8')
+      ) as InstallPlanConfig,
+    }));
+    const installPlan = allInstallPlans.find((i) => i.content?.id === id);
+    this.localPath = path.dirname(
+      removeBasePath(
+        installPlan?.filePath ?? '',
+        path.join(this.basePath, 'install')
+      )
+    );
+
+    const filePaths = glob.sync(
+      path.join(this.basePath, 'install', this.localPath, 'install.+(yml|yaml)')
+    );
+
+    if (!Array.isArray(filePaths) || filePaths.length !== 1) {
+      this.isValid = false;
+      return '';
+    }
+
+    return removeBasePath(filePaths[0], this.basePath);
+  }
+
+  getConfigContent() {
+    return this._getYamlConfigContent();
+  }
+
+  /**
+   * Get the variables for the **Quickstart** mutation.
+   *
+   * @returns The ID for this install plan
+   */
+  getMutationVariables() {
+    return this.config.id;
+  }
+
+  /**
+   * Submits a mutation to NerdGraph for a single install plan.
+   */
+  public async submitMutation(dryRun = true) {
+    const { data, errors } = await fetchNRGraphqlResults<
+      InstallPlanMutationVariable,
+      InstallPlanMutationResponse
+    >({
+      queryString: INSTALL_PLAN_MUTATION,
+      variables: this._getComponentMutationVariables(dryRun),
+    });
+
+    return { name: this.config.name, data, errors };
+  }
+
+  /**
+   * Get the **component-specific** mutation variables.
+   *
+   * @todo: add this to the abstract class (or create an interface)
+   */
+  private _getComponentMutationVariables(
+    dryRun: boolean
+  ): InstallPlanMutationVariable {
+    const { id, description, name, title, install, fallback, target } =
+      this.config ?? {};
+
+    return {
+      dryRun,
+      id,
+      description,
+      displayName: name,
+      heading: title,
```

Comment:
Let's trim these to be safe
```suggestion
      displayName: name && name.trim(),
      heading: title && title.trim(),
```